app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt

```diff
@@ -431,8 +435,8 @@ class SettingsFragment : PreferenceFragmentCompat() {
         val debuggingKey = getPreferenceKey(R.string.pref_key_remote_debugging)
         val preferenceLeakCanary = findPreference<Preference>(leakKey)
         val preferenceRemoteDebugging = findPreference<Preference>(debuggingKey)
-        val preferenceMakeDefaultBrowser =
-            requirePreference<Preference>(R.string.pref_key_make_default_browser)
+        preferenceMakeDefaultBrowser =
```

Comment:
```suggestion
val preferenceMakeDefaultBrowser =
    requirePreference<DefaultBrowserPreference>(R.string.pref_key_make_default_browser)
```

Don't think we need to cache the preference and update it in other methods.
The proposed flow would have the value updated in `onResume` but that method already calls `update()` which calls this method - `setupPreferences()`. So it looks like here is the place where we'd need to ensure all preferences are correctly configured and displayed.

Seems like below a click listener is set for the default browser preference so that call can be updated to something similar to:
```kotlin
preferenceMakeDefaultBrowser.apply {
    updateSwitch()
    onPreferenceClickListener =
        getClickListenerForMakeDefaultBrowser()
}
```

Reply:
Done